coveo-functools/coveo_functools/flex/types.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from typing import Any
+
+
+JSON_TYPES = (
+    str,
+    bool,
+    int,
+    float,
+    type(None),
+    dict,
+)  # list omitted to support list of custom types
+
+PASSTHROUGH_TYPES = {None, Any, *JSON_TYPES}
+
+
+TypeHint = Any  # :shrug:

```

Comment:
This was all moved from `flex.py` and put in here so not to induce any circular references.